### ISSUE TITLE
not reporting syntax errors if I use the "select" option.

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#69]
In GitLab by @cakoose on Jan 16, 2015, 01:35

The docs for the "select" config option say "select errors and warnings to enable which are off by default".  This seems to indicate that "select" can only increase the set of errors reported, but I'm seeing otherwise.

By default, syntax errors are reported:

```
$ cat test.py
def f(x)
    print x
$ flake8 test.py
test.py:1:10: E901 SyntaxError: invalid syntax
test.py:2:5: E113 unexpected indentation
$
```

But let's say I have a "setup.cfg" file in the same folder.  Using the "select" option causes unexpected behavior:

```
$ cat setup.cfg
[flake8]
select = E101
$ flake8 test.py
$
```

```
$ cat setup.cfg
[flake8]
select = E901
$ flake8 test.py
$
```

```
$ cat setup.cfg
[flake8]
select = E113
$ flake8 test.py
test.py:2:5: E113 unexpected indentation
$
```

Version: 2.3.0 (pep8: 1.5.7, pyflakes: 0.8.1, mccabe: 0.3) CPython 2.7.5 on Darwin

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jan 16, 2015, 14:01

No. `select` is behaving exactly as it should there. `select` selects which errors you want to see. That way if you don't want to see 1 error, you just ignore it, but if you want to focus on fixing instances of 1 error, you select it.

**Comment by asottile** [comment]
In GitLab by @cakoose on Jan 16, 2015, 14:53

1. Flake8 doesn't report the E901-category error even when "select = E901" (the second-to-last example in my original comment).  Should I file this as a separate bug?

2. The docs for "select" says: "select errors and warnings to enable which are off by default".  It's the "which are off by default" part that threw me off.  A better doc: "only enable the errors and warnings listed here".

3. Is there a way to tell Flake8 "enable all the standard errors and warnings, plus these few specific warnings I choose"?

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jan 24, 2015, 11:31

> Flake8 doesn't report the E901-category error even when "select = E901" (the second-to-last example in my original comment). Should I file this as a separate bug?

E901 is not an error code I'm aware of. Do you mean C901? If C901 is what you mean *and* C901 is what you're using, then you should be using `--max-complexity` to enable that.

> The docs for "select" says: "select errors and warnings to enable which are off by default". It's the "which are off by default" part that threw me off. A better doc: "only enable the errors and warnings listed here"

There's a list under [Default](https://flake8.readthedocs.org/en/2.2.3/config.html#default)

> Is there a way to tell Flake8 "enable all the standard errors and warnings, plus these few specific warnings I choose"?

No. There is no way to do that in pep8 so there is no way to do it in flake8 (since flake8 merely adds extension capability and a few extra defaults on top of pep8 at this point).

**Comment by asottile** [comment]
In GitLab by @cakoose on Jan 25, 2015, 01:32

See the first example in the bug report.  E901 is what it says for the syntax error.

Re: the docs.  What does the "which are off by default" mean?  Why shouldn't that phrase just be deleted from that sentence?

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jan 25, 2015, 09:12

Re the first example, here's a corresponding bug report: https://gitlab.com/pycqa/flake8/issues/30

> What does the "which are off by default" mean? Why shouldn't that phrase just be deleted from that sentence?

There are error codes that are ignored (a.k.a., off, not reported, etc.) by default. It is pep8's default list of ignored error codes/classes.
